precompose/src/commonMain/kotlin/moe/tlaster/precompose/navigation/BackStackManager.kt

```diff
@@ -49,10 +54,16 @@ internal class BackStackManager : LifecycleObserver {
     fun init(
         routeGraph: RouteGraph,
         stateHolder: StateHolder,
-        lifecycleOwner: LifecycleOwner,
+        savedStateHolder: SavedStateHolder,
+        lifecycleOwner: LifecycleOwner
     ) {
         _stateHolder = stateHolder
+        _savedStateHolder = savedStateHolder
         lifecycleOwner.lifecycle.addObserver(this)
+
+        _savedStateHolder.registerProvider(STACK_SAVED_STATE_KEY) {
+            backStacks.value.map { backStackEntry -> backStackEntry.path }
```

Comment:
Can you add an option to opt-in this feature? Not all apps want to restore the backstacks after being killed.

Reply:
Since this library is intended as a replacement of the Jetpack Navigation just for multiplatform, shouldn't it behave like it on this matter? unless I'm mistaken, Jetpack Navigation [restores](https://github.com/androidx/androidx/blob/006680832f0d7fc8d31a05ab20ab217cc29101d1/navigation/navigation-compose/src/main/java/androidx/navigation/compose/NavHostController.kt#L78-L83) navigation and there is no way for apps to opt-out.

This is just to discuss, but I can apply the change if you prefer, and if you do, WDYT is the best place for the opt-out, as a parameter to `rememberNavigator` or in the `NavHost`?

Reply:
As I'll be busy next week, I implemented your suggestion in 83d07c1, if you prefer another approach, feel free to fork the branch and make any adjustments, otherwise I can make the adjustments later.

Reply:
Thanks for the update, it's now LGTM!